url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/io/Generic_StaticIO.java

```diff
@@ -0,0 +1,2156 @@
+     */
+    public static void skipline(StreamTokenizer streamTokenizer) {
+        int tokenType;
+        try {
+            tokenType = streamTokenizer.nextToken();
+            while (tokenType != StreamTokenizer.TT_EOL) {
+                tokenType = streamTokenizer.nextToken();
+            }
+        } catch (IOException ex) {
+            Logger.getLogger(Generic_StaticIO.class.getName()).log(
+                    Level.SEVERE, null, ex);
+        }
+    }
+
+    /**
+     * Sets the syntax of st as follows:
+     * @code {
+     * <ul>
+     * <li>st.wordChars( '0', '0' );</li>
+     * <li>st.wordChars( '1', '1' );</li>
+     * <li>st.wordChars( '2', '2' );</li>
+     * <li>st.wordChars( '3', '3' );</li>
+     * <li>st.wordChars( '4', '4' );</li>
+     * <li>st.wordChars( '5', '5' );</li>
+     * <li>st.wordChars( '6', '6' );</li>
+     * <li>st.wordChars( '7', '7' );</li>
+     * <li>st.wordChars( '8', '8' );</li>
+     * <li>st.wordChars( '9', '9' );</li>
+     * </ul>
+     * }
+     *
+     * @param st <code>StreamTokenizer</code> thats syntax is set.
+     */
+    public static void setStreamTokenizerSyntaxNumbersAsWords1(
+            StreamTokenizer st) {
+        // st.wordChars( '0', '9' );
+        st.wordChars('0', '0');
+        st.wordChars('1', '1');
+        st.wordChars('2', '2');
+        st.wordChars('3', '3');
+        st.wordChars('4', '4');
+        st.wordChars('5', '5');
+        st.wordChars('6', '6');
+        st.wordChars('7', '7');
+        st.wordChars('8', '8');
+        st.wordChars('9', '9');
+    }
+
+    /**
+     * Sets the syntax of st as follows:
+     * @code {
+     * <ul>
+     * <li>st.resetSyntax();</li>
+     * <li>st.wordChars( ',', ',' );</li>
+     * <li>st.wordChars( '"', '"' );</li>
+     * <li>setStreamTokenizerSyntaxNumbersAsWords1(st);</li>
+     * <li>st.wordChars( '.', '.' );</li>
+     * <li>st.wordChars( '-', '-' );</li>
+     * <li>st.wordChars( '+', '+' );</li>
+     * <li>st.wordChars( 'a', 'z' );</li>
+     * <li>st.wordChars( 'A', 'Z' );</li>
+     * <li>st.wordChars( '\t', '\t' );</li>
+     * <li>st.wordChars( ' ', ' ' );</li>
+     * <li>st.eolIsSignificant( true );</li>
+     * </ul>
+     * }
+     *
+     * @param st <code>StreamTokenizer</code> thats syntax is set.
+     */
+    public static void setStreamTokenizerSyntax1(
+            StreamTokenizer st) {
+        st.resetSyntax();
+        // st.parseNumbers();
+        st.wordChars(',', ',');
+        st.wordChars('"', '"');
+        // st.whitespaceChars( '"', '"' );
+        setStreamTokenizerSyntaxNumbersAsWords1(st);
+        st.wordChars('.', '.');
+        st.wordChars('-', '-');
+        st.wordChars('_', '_');
+        st.wordChars('+', '+');
+        st.wordChars('a', 'z');
+        st.wordChars('A', 'Z');
+        // char[] tab = new char[1];
+        // tab[0] = '\t';
+        st.wordChars('\t', '\t');
+        st.wordChars(' ', ' ');
+        // st.ordinaryChar( ' ' );
+        st.eolIsSignificant(true);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 836:853, 915:932.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.